Sources/Tentacle/Client.swift

```diff
@@ -370,7 +372,13 @@ public final class Client {
 
     /// Fetch the tree for a repository reference
     public func tree(in repository: Repository, atRef ref: String = "HEAD", recursive: Bool = false) -> SignalProducer<(Response, Tree), Error> {
-        return fetchOne(.tree(owner: repository.owner, repository: repository.name, ref: ref, recursive: recursive))
+        return fetchOne(.tree(owner: repository.owner, repository: repository.name, recursive: recursive, ref: ref))
+    }
+
+    /// Create a tree in a repository
+    public func create(tree: [Tree.Entry], basedOn base: String?, in repository: Repository, inBranch branch: String? = nil) -> SignalProducer<(Response, FileResponse), Error> {
+        let newTree = NewTree(entries: tree, base: base)
+        return send(newTree, to: .tree(owner: repository.owner, repository: repository.name, recursive: false, ref: nil), using: .post)
```

Comment:
Actually, the branch argument should not be there. The `tree` endpoint is used for both getting and creating a tree, where the path takes the following formats respectively:

- `/repos/:owner/:repo/git/trees/:sha`
- `/repos/:owner/:repo/git/trees`

when [creating the tree](https://developer.github.com/v3/git/trees/#create-a-tree), the base is part of the `NewTree`, rather than a ref for `Endpoint.tree`. 

We could create a new `Endpoint` `createTree` if you think this would be clearer.